blacklist/README.md

```diff
@@ -0,0 +1,16 @@
+# file extend name
+`.json`
+# File format
+```
+# case blacklist/custom1.json
+{
+    "platform": "codename",
+    "driver": "driver",
+    "blacklist": {
+        "test/suite":{
+            "case1":"reason",
+            "case2":"reason"
+            }
+        }
+}
```

Comment:
To avoid extra parsing in code, just impose structure like:
```json
{
  "platform:driver": {
    "test/suite": {
      "case1": "reason",
      "case2": "reason"
    }
}
```

Reply:
Also, I'm not sure how `self.blacklist.update(...)` would behave when there are nested dictionaries with duplicate top-level keys.  So maybe it's better to only allow a single blacklist.json file per run.

Reply:
What about the case variations?  We might want to blacklist CASE_XYZ only if the variation includes `bframes=3`, etc...

Reply:
> What about the case variations? We might want to blacklist CASE_XYZ only if the variation includes `bframes=3`, etc...

This is why I suggested we use the same filtering syntax and mechanism as slash uses for the `-k` option.  Look at the slash code to figure out how.

Reply:
> > What about the case variations? We might want to blacklist CASE_XYZ only if the variation includes `bframes=3`, etc...
> 
> This is why I suggested we use the same filtering syntax and mechanism as slash uses for the `-k` option. Look at the slash code to figure out how.

We use a slash matcher in https://github.com/intel/vaapi-fits/blob/master/tools/bdrate

Reply:
> I already try the `update`, it is merge the value
> or we just support one file?
> or direct load target blacklist file?
> if continue use multiple file, it still need `self.blacklist["platform:driver"]["test/suite"]` to keep effect range

I think we support just one file per run... it simplifies things.

Reply:
Sorry, the top-level json structure does not need to be a list… it should be the first inner dictionary in my last example

Reply:
Here is a generic function that we can call within the scope of test_start()... hope this helps:

```python
  def _assert_not_matches(self, test, filter, message):
    from slash.utils.pattern_matching import Matcher
    assert not Matcher(filter).matches(test.__slash__), message
```

Reply:
We can make filtering even more powerful, extendable and generic with json like:
```json
{
  "filter1" : "reason1",
  "filter2" : "reason2"
}
```
...where filter includes things like `platform=...` and `driver=...`.  This would allow one rule to cover multiple drivers and/or platforms in a single rule.

...then we just include driver and platform automatically in search string:

```python
  def _assert_not_matches(self, test, filter, message):
    from slash.utils.pattern_matching import Matcher
    strs = [
      str(test.__slash__),
      "driver=" + self._get_driver_name(),
      "platform=" + self._get_platform_name(),
    ]
    assert not Matcher(filter).matches(" ".join(strs)), message
```
... thus, now we can define a filter like `"ffmpeg-vaapi/transcode/avc and case=MYCASE and (platform=TGL or platform=KBL) and driver=iHD"`